target_chains/ethereum/contracts/contracts/pulse/Pulse.sol

```diff
@@ -171,10 +190,15 @@ abstract contract Pulse is IPulse, PulseState {
     }
 
     function getFee(
-        uint256 callbackGasLimit
+        uint256 callbackGasLimit,
+        address provider
     ) public view override returns (uint128 feeAmount) {
+        if (provider == address(0)) {
+            provider = _state.defaultProvider;
+        }
         uint128 baseFee = _state.pythFeeInWei;
-        uint256 gasFee = callbackGasLimit * tx.gasprice;
+        uint128 providerFeeInWei = _state.providers[provider].feeInWei;
+        uint256 gasFee = callbackGasLimit * providerFeeInWei;
```

Comment:
as discussed, changed this to use provider fee to protect from future price jumps/drops